src/apiCall.js

```diff
@@ -1,23 +1,24 @@
-const pushtx = require('blockchain.info/pushtx');
-const blockexplorer = require('blockchain.info/blockexplorer');
-const request = require("request");
-
+const request = require('request');
 
 module.exports = {
     getData: (endPoint) => {
-        return new Promise( (fulfill, reject) => { // Create Promise
+        return new Promise((fulfill, reject) => { // Create Promise
             request({
                 method: 'GET',
                 url: endPoint,
                 headers: {
-                    'Content-Type': 'application/json',
-                },
-            },  (error, response, body) => {
-                if (error) {
-                    reject(error);
+                    'Content-Type': 'application/json'
                 }
-                else {
-                    fulfill(JSON.parse(body));
+            },  (error, response, body) => {
+                if (error)
+                    return reject(error);
+
+                try {
+                    const result = JSON.parse(body);
```

Comment:
Without the `try/catch` the error thrown by `JSON.parse` would bubble up and end the process.